packages/zoe/src/types-ambient.js

```diff
@@ -13,26 +13,42 @@
 /**
  * @typedef {string} Keyword
  * @typedef {Handle<'Invitation'>} InvitationHandle - an opaque handle for an invitation
- * @typedef {Record<Keyword, Issuer<any>>} IssuerKeywordRecord
- * @typedef {Record<Keyword, ERef<Issuer<any>>>} IssuerPKeywordRecord
- * @typedef {Record<Keyword, Brand<any>>} BrandKeywordRecord
  */
 
 /**
+ * @template {Record<Keyword, AssetKind>} [IKR=Record<Keyword, any>]
+ * @typedef {{ [K in keyof IKR]: Issuer<IKR[K]> }} IssuerKeywordRecord
+ */
+
+/**
+ * @template {Record<Keyword, AssetKind>} [IKR=Record<Keyword, any>]
+ * @typedef {{ [K in keyof IKR]: Brand<IKR[K]> }} BrandKeywordRecord
+ */
+
+/**
+ * @template {Record<Keyword, AssetKind>} [IKR=Record<Keyword, any>]
+ * @typedef {{ [K in keyof IKR]: ERef<Issuer<IKR[K]>> }} IssuerPKeywordRecord
+ */
+
+/**
+ * @template {Record<Keyword, AssetKind>} [IKR=Record<Keyword, any>]
  * @typedef {object} StandardTerms
- * @property {IssuerKeywordRecord} issuers - record with
+ * @property {IssuerKeywordRecord<IKR>} issuers - record with
  * keywords keys, issuer values
- * @property {BrandKeywordRecord} brands - record with keywords
+ * @property {BrandKeywordRecord<IKR>} brands - record with keywords
```

Comment:
this one will take a little type-fu to make a BrandKeywordRecord from an IssuerKeywordRecord. that can be a utility type.

the upside is that `brands` will be the actual brands